lib/livebook_cli/deploy.ex

```diff
@@ -121,9 +121,15 @@ defmodule LivebookCLI.Deploy do
     log_debug("Authenticating CLI...")
 
     case Teams.fetch_cli_session(config) do
-      {:ok, team} -> team
-      {:error, error} -> raise LivebookCLI.Error, error
-      {:transport_error, error} -> raise LivebookCLI.Error, error
+      {:ok, team} ->
+        team
+
+      {:error, error} ->
+        %{"errors" => %{"detail" => error_message}} = error
+        raise LivebookCLI.Error, error_message
+
+      {:transport_error, error} ->
+        raise LivebookCLI.Error, error
     end
```

Comment:
Since the function `Teams.fetch_cli_session/1` expects to return a `{:error, String.t()}`, the Teams function could pattern match on this kind of error and return the error message directly.